source/gamemap/MiniMapDrawnFull.cpp

```diff
@@ -502,3 +502,142 @@ void MiniMapDrawnFull::updateTileState(uint32_t minimapXMin, uint32_t minimapXMa
     colourFromPixelValue(curValue, seatIfClaimed, mPixelBuffer, mPixelBox,
         mWidth, mHeight, minimapXMin, minimapXMax, minimapYMin, minimapYMax);
 }
+
+bool MiniMapDrawnFull::crossSegment(const Ogre::Vector3& p1, const Ogre::Vector3& p2,
+        uint32_t xMin, uint32_t xMax, uint32_t yMin, uint32_t yMax)
+{
+    if((p1.x < static_cast<Ogre::Real>(xMin)) &&
+       (p2.x < static_cast<Ogre::Real>(xMin)))
+    {
+        return false;
+    }
+    if((p1.x > static_cast<Ogre::Real>(xMax)) &&
+       (p2.x > static_cast<Ogre::Real>(xMax)))
+    {
+        return false;
+    }
+
+    if((p1.y < static_cast<Ogre::Real>(yMin)) &&
+       (p2.y < static_cast<Ogre::Real>(yMin)))
+    {
+        return false;
+    }
+    if((p1.y > static_cast<Ogre::Real>(yMax)) &&
+       (p2.y > static_cast<Ogre::Real>(yMax)))
+    {
+        return false;
+    }
+
+    Ogre::Real diffYPoints = p2.y - p1.y;
+    Ogre::Real diffXPoints = p2.x - p1.x;
+    Ogre::Real diffYMin = p2.y - static_cast<Ogre::Real>(yMin);
+    Ogre::Real diffXMin = p2.x - static_cast<Ogre::Real>(xMin);
+    Ogre::Real diffYMax = p2.y - static_cast<Ogre::Real>(yMax);
+    Ogre::Real diffXMax = p2.x - static_cast<Ogre::Real>(xMax);
+
+    // Magic number to change the size of the line
+    static const Ogre::Real DIFF_MIN = 5;
+    if(
+       (
+        (diffYMin * diffXPoints - diffYPoints * diffXMin - DIFF_MIN< 0) &&
```

Comment:
Well, honestly, I think both are hard to read so I don't tink it is worth the trouble if this is the only thing to change